### DISCUSSION TITLE
benchmark inputs

### DISCUSSION THREAD

**Discussion by sroy729** [sarchlab/mgpusim#75] [Q&A] [answered]
Does the sample code of benchmark in `sample` have the standard input? like for example, in Heterobenchmark, it shows the following input 
```
Workload          Input
AES               A plaintext of 1048576 bytes
BE                 One second (24 frames) 1080p
                     (1920 × 1080 pixels) B&W video
BS                65536 stock price samples
CH               A 3000 × 2000 pixels B&W picture
EP                3 generations, each generation
                    has 10000 population
FIR              100 groups (32768 samples per group)
GA               Search an 1024 element sequence
                     against an 1M element sequence
KM              10000 34 dimension data points
PR                4096 Nodes
```
another question, in general, how do you decide the input when running a benchmark for GPU? Like in CPU, there are SPEC benchmark suits that have most of the things predefined; what similar thing to do when working with GPUs?

**Comment by syifan** [comment, marked as answer]
No, they are pretty much randomly generated. 

For the workloads, my recommendation is to have the smallest problem sizes, where each problem size has a sufficient number of threads that fully occupy the GPU a few times. Another way is to find the problem size that each simulation runs between 6-12 hours.
